src/wallet/orchard.h

```diff
@@ -124,10 +129,31 @@ class OrchardWallet
                 &result,
                 PushOrchardActionIVK
                 );
-
         return result;
     }
 
+    /**
+     * Decrypts a selection of notes from the specified transaction's
+     * Orchard bundle with provided incoming viewing keys, and adds those
+     * notes to the wallet.
+     */
+    bool RestoreDecryptedNotes(
+            const CTransaction& tx,
+            std::map<uint32_t, libzcash::OrchardIncomingViewingKey> hints
+            ) {
+        std::vector<RawOrchardActionIVK> rawHints;
```

Comment:
I am ✨ W A R Y ✨  of this usage; while I'm confident that `std::vector<T>` is allocated as contiguous memory (in C++20 it in fact implements an interface defining it as such), I'm not confident that the `RawOrchardActionIVK` struct won't be padded. It's a 12-byte value, so with an alignment of 4 or less we _should_ be fine? But I can't find anywhere that ensures this (vs e.g. `std::vector<uint8_t>` which will have an alignment of 1, and thus is fine to pass across the FFI).

Let's add static assertions that check the size and alignment of `RawOrchardActionIVK`, like we do for `HistoryEntry`.